### PR TITLE
Completed rps challenge 

### PR DESCRIPTION
**Pull request by micosmin**
Completed bonus rounds, but no time to make it pretty at this stage

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
 
-* Challenge time: rest of the day and weekend, until Monday 9am
-* Feel free to use google, your notes, books, etc. but work on your own
-* If you refer to the solution of another coach or student, please put a link to that in your README
-* If you have a partial solution, **still check in a partial solution**
-* You must submit a pull request to this repo with your code by 9am Monday morning
-
-Task
-----
+## Task
```
Comment (amyj0rdan) [inline README.md]: Add information on how to run the program in your README (eg, rackup, ruby app.rb, etc.)

File: lib/computer.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Computer < Player
```
Comment (amyj0rdan) [inline lib/computer.rb]: Nice!

File: README.md [hunk 3]
```diff
@@ -36,12 +28,60 @@ Hints on functionality
...
-* TEST driving development of your app
-
+## Approach to solving the challenge
+
+1. Forked repo
+2. Installed dependencies
+3. Set up Travis to follow the rps-challenge repo
+4. Added travis status badge to README
+5. Set up spec_helper to work with capybara
+6. Set up app.rb and config.ru
```
Comment (amyj0rdan) [inline README.md]: According to rubric, we should have an app folder that includes views as a subfolder

File: lib/computer.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  def generate_computer_choice(game_options)
+    @choice = game_options[rand(1..game_options.count)]
+  end
+
+  def computer_choice
+    @choice
+  end
+
+end
```
Comment (amyj0rdan) [inline lib/computer.rb]: Add unit tests for the Computer class?

File: lib/game_options.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class GameOptions
```
Comment (amyj0rdan) [inline lib/game_options.rb]: Add unit tests for the GameOptions class?

File: spec/unit/winner_spec.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require 'winner'
```
Comment (amyj0rdan) [inline spec/unit/winner_spec.rb]: Test all possible game logic (rock v paper, paper v scissors, scissors v rock - and the reverse)

File: spec/unit/winner_spec.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (amyj0rdan) [inline spec/unit/winner_spec.rb]: Add scenario and tests for a draw

File: spec/features/play_game_spec.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  scenario 'go back to play screen after start again button is clicked' do
+    register_1_player_against_computer
+    click_button 'rock'
+    click_link 'OK'
+    click_button 'Start again'
+    expect(page).to have_content 'Round 2'
+  end
+
+  scenario 'shows score increase for player 1' do
+    allow_any_instance_of(Computer).to receive(:rand).and_return(2)
```
Comment (amyj0rdan) [inline spec/features/play_game_spec.rb]: Nice! Could also have done (per rubric) 'allow_any_instance_of(Array).to receive(:sample).and_return('rock')' and then tests will be easier to read to see what the output of the rand will be.

File: spec/unit/winner_spec.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+describe Winner do
+  subject(:winner) { described_class }
+  let(:p1) { double :player1, :name => 'p1', :user_choice => 'Rock', :increase_score => 0 }
+  let(:p2) { double :player2, :name => 'p2', :user_choice => 'Paper', :increase_score => 1 }
+
+  it '#declare_winner' do
```
Comment (amyj0rdan) [inline spec/unit/winner_spec.rb]: Nice that it returns the object of the winner rather than a string. Might want it to return a win object or a draw object to allow for a draw scenario.

File: lib/game_options.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+  def self.normal_game
```
Comment (amyj0rdan) [inline lib/game_options.rb]: Could these be stored in a constant? Although this is only called once at the start of a session, if that changed and was called again later, would your program recreate these objects?

File: spec/features/play_game_spec.rb [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  scenario 'player 2 wins game' do
+    allow_any_instance_of(Computer).to receive(:rand).and_return(2)
+    register_1_player_against_computer
+    click_button 'rock'
+    click_link 'OK' 
+    expect(page).to have_content 'Winner: Computer'
+  end
+
```
Comment (amyj0rdan) [inline spec/features/play_game_spec.rb]: Add draw scenario

File: spec/features/register_name_spec.rb [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# As a marketeer
+# So that I can see my name in lights
+# I would like to register my name before playing an online game
+require './app'
+
+feature 'Start game' do
```
Comment (amyj0rdan) [inline spec/features/register_name_spec.rb]: Test for if they want to start a spock game?

File: spec/features/register_name_spec.rb [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# As a marketeer
```
Comment (amyj0rdan) [inline spec/features/register_name_spec.rb]: Test for edge cases (eg, if the user doesn't enter any information?)

File: spec/features/select_game_type_spec.rb [hunk 14]
```diff
@@ -0,0 +1,17 @@
+feature 'Game has 2 different play options' do
+  
+  scenario 'user can select normal game' do
+    register_2_players_and_start_game
+    expect(page).to have_content 'Game type normal'
+    expect(page).not_to have_button 'spock'
+    expect(page).not_to have_button 'lizzard'
```
Comment (amyj0rdan) [inline spec/features/select_game_type_spec.rb]: Typo - 'lizard' :)

File: spec/features/show_score_spec.rb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+feature 'Game keeps score' do
+  scenario 'player score changes after first round' do
+    visit '/'
+    register_2_players_and_start_game
+    click_button 'paper'
+    click_button 'rock'
+    click_button 'Start again'
+    expect(page).to have_content 'Score: Player1 - 1 : Player2 - 0'
+  end
```
Comment (amyj0rdan) [inline spec/features/show_score_spec.rb]: Add a test for more rounds?

File: spec/features/web_helpers_spec.rb [hunk 16]
```diff
@@ -0,0 +1,39 @@
+def register_2_players_and_start_game
```
Comment (amyj0rdan) [inline spec/features/web_helpers_spec.rb]: Nice extraction!

File: lib/winner.rb [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+
+  def declare_winner
+    p1_choice = @p1.user_choice.to_sym
+    p2_choice = @p2.user_choice.to_sym
+    compute_winner(p1_choice, p2_choice)
+  end
+
+  private
+
+  def compute_winner(p1_choice, p2_choice)
```
Comment (amyj0rdan) [inline lib/winner.rb]: This is a long method. Ok as it is private? Could use a hash with the rules

File: views/chose_game_type.erb [hunk 18]
```diff
@@ -0,0 +1,8 @@
+<h1>Select game type</h1>
+
+<form action="/create_game" method='post'>
+
+<button type="submit" name="game_type" value="normal"><img src="/images/rps.png"></button>
```
Comment (amyj0rdan) [inline views/chose_game_type.erb]: Cool that you have button images :)
Maybe add some more snazzy CSS!
